shared/aggregation/maxcover.go

```diff
@@ -79,11 +78,6 @@ func (mc *MaxCoverProblem) Cover(k int, allowOverlaps bool) (*Aggregation, error
 				break
 			}
 			if !candidate.processed {
-				if !allowOverlaps && solution.Coverage.Overlaps(*candidate.bits) {
-					// Overlapping candidates violate non-intersection invariant.
-					candidate.processed = true
-					continue
-				}
```

Comment:
This code is redundant, as above (on line 74), we filter out overlapping candidates in:
```golang
// Score candidates against remaining bits.
// ***Filter out processed and overlapping (when disallowed).***
// Sort by score in a descending order.
mc.Candidates.score(remainingBits).filter(solution.Coverage, allowOverlaps).sort()
```